src/main/java/christmas/domain/date/DecemberCalendar.java

```diff
@@ -0,0 +1,93 @@
+    DAY_1(new VisitDate("1"), false, true, false),
+    DAY_2(new VisitDate("2"), false, true, false),
+    DAY_3(new VisitDate("3"), true, false, true),
+    DAY_4(new VisitDate("4"), true, false, false),
+    DAY_5(new VisitDate("5"), true, false, false),
+    DAY_6(new VisitDate("6"), true, false, false),
+    DAY_7(new VisitDate("7"), true, false, false),
+    DAY_8(new VisitDate("8"), false, true, false),
+    DAY_9(new VisitDate("9"), false, true, false),
+    DAY_10(new VisitDate("10"), true, false, true),
+    DAY_11(new VisitDate("11"), true, false, false),
+    DAY_12(new VisitDate("12"), true, false, false),
+    DAY_13(new VisitDate("13"), true, false, false),
+    DAY_14(new VisitDate("14"), true, false, false),
+    DAY_15(new VisitDate("15"), false, true, false),
+    DAY_16(new VisitDate("16"), false, true, false),
+    DAY_17(new VisitDate("17"), true, false, true),
+    DAY_18(new VisitDate("18"), true, false, false),
+    DAY_19(new VisitDate("19"), true, false, false),
+    DAY_20(new VisitDate("20"), true, false, false),
+    DAY_21(new VisitDate("21"), true, false, false),
+    DAY_22(new VisitDate("22"), false, true, false),
+    DAY_23(new VisitDate("23"), false, true, false),
+    DAY_24(new VisitDate("24"), true, false, true),
+    DAY_25(new VisitDate("25"), true, false, true),
+    DAY_26(new VisitDate("26"), true, false, false),
+    DAY_27(new VisitDate("27"), true, false, false),
+    DAY_28(new VisitDate("28"), true, false, false),
+    DAY_29(new VisitDate("29"), false, true, false),
+    DAY_30(new VisitDate("30"), false, true, false),
+    DAY_31(new VisitDate("31"), true, false, true);
+
+    private final VisitDate date;
+    private final boolean weekday;
+    private final boolean weekend;
+    private final boolean star;
+
+    DecemberCalendar(final VisitDate date, final boolean weekday, final boolean weekend, final boolean star) {
+        this.date = date;
+        this.weekday = weekday;
+        this.weekend = weekend;
+        this.star = star;
+    }
+
+    public static boolean matchWeekday(final VisitDate visitDate) {
+        int date = visitDate.getDate();
+        for (DecemberCalendar day : DecemberCalendar.values()) {
+            if (day.getDate() == date) {
+                return day.isWeekday();
+            }
+        }
+        return false;
+    }
+
+    public static boolean matchWeekend(final VisitDate visitDate) {
+        int date = visitDate.getDate();
+        for (DecemberCalendar day : DecemberCalendar.values()) {
+            if (day.getDate() == date) {
+                return day.isWeekend();
+            }
+        }
+        return false;
+    }
+
+    public static boolean matchStar(final VisitDate visitDate) {
+        int date = visitDate.getDate();
+        for (DecemberCalendar day : DecemberCalendar.values()) {
+            if (day.getDate() == date) {
+                return day.isStar();
+            }
+        }
+        return false;
+    }
```

Comment:
전달받은 `visitDate`에 따라 전체 `values()`를 순회하는 로직이 중복하여 존재하는데요.
이 부분을 따로 추출하여 `visitDate`에 매핑되는 `DecemberCalendar`를 찾는 로직을 구현한다면 이 부분들이 훨씬 간결해 질 것 같아요


```suggestion
    public static DecemberCalendar from(final VisitDate visitDate){
        return Arrays.stream(values())
                .filter(day -> day.getDate() == visitDate.getDate())
                .findFirst()
                .orElseThrow(IllegalArgumentException::new);
    }
    
    public static boolean matchWeekday(final VisitDate visitDate) {
        return from(visitDate).isWeekday();
    }

    public static boolean matchWeekend(final VisitDate visitDate) {
        return from(visitDate).isWeekend();
    }

    public static boolean matchStar(final VisitDate visitDate) {
        return from(visitDate).isStar();
    }
```

Reply:
저번 리뷰때도 느낀 건데 장수님은 Java8의 Stream을 정말 잘 활용하시는 것 같아요 한 수 배우갑니다!